src/session.ts

```diff
@@ -528,10 +531,6 @@ async function warmTransfer(
     target: WebPhoneSession,
     transferOptions: any = {}
 ): Promise<ReferClientContext> {
-    await (this.localHold ? Promise.resolve(null) : this.hold());
-
-    await delay(300);
-
```

Comment:
> AFAIR warmTransfer requires that call should be on hold. Is it a safe fix? Has anything changed in telco?
>
> The issue is that hold might not be finished in 300ms, this logic trigger a bug in Jupiter when app send REFER before the reInvite is success, server will return success for the REFER but do nothing.
> 
> In mThor, we won't hold call before we transfer / warmTransfer it. I also capture the SIP for Spartan and spartan will also NOT hold call before transfer call

Hold is needed because w/o it other party will hear that something is going on. Cisco and Polycom do it. Alternatively you can mute yourself. Looks like we need to implement a better hold wait method, per telco: **Refer does work w/o hold. Hold is necessary to allow transferor to make a consult call before the transfer**

Reply:
We ran into issues earlier when we used Refer for transfer , i donot know if that has chnaged, we can check with telco

Reply:
@vyshakhbabji figure out how it's used in dThor